server/resourcemanager/testutil/operations_list.go

```diff
@@ -1,109 +1,251 @@
-	return buildListRequest(
-		rt.ResourceType,
-		ct,
-		testServer,
-		t,
-	)
+const OperationListPaginatedAscendingSuccess Operation = "ListPaginatedAscendingSuccess"
+
+var listPaginatedAscendingSuccessOperation = operationTester{
+	name: OperationListPaginatedAscendingSuccess,
+	buildRequest: func(t *testing.T, testServer *httptest.Server, ct contentTypeConverter, rt ResourceTypeTest) *http.Request {
+		return buildListRequest(
+			rt.ResourceType,
+			map[string]string{
+				"take":          "2",
+				"skip":          "1",
+				"sortBy":        getSortFieldsString(rt.PaginationSortFields),
+				"sortDirection": "asc",
+			},
+			ct,
+			testServer,
+			t,
+		)
+	},
+	assertResponse: func(t *testing.T, resp *http.Response, ct contentTypeConverter, rt ResourceTypeTest) {
+		require.Equal(t, 200, resp.StatusCode)
+
+		jsonBody := responseBodyJSON(t, resp, ct)
+
+		var parsedJsonBody struct {
+			Count int              `json:count`
+			Items []map[string]any `json:items`
+		}
+		json.Unmarshal([]byte(jsonBody), &parsedJsonBody)
+
+		require.Equal(t, 3, parsedJsonBody.Count)
+
+		var prevVal any
+
+		sortFields := getSafeSortFields(rt.PaginationSortFields)
+
+		for _, field := range sortFields {
+			for _, item := range parsedJsonBody.Items {
+				if prevVal == nil {
+					prevVal = item[field]
+					continue
+				}
+				assert.True(t, compareFieldsLesserThan(prevVal, item[field]))
+
+				prevVal = item[field]
+			}
+		}
+	},
 }
 
-func (listInteralErrorOperation) name() Operation {
-	return OperationListInteralError
+const OperationListPaginatedDescendingSuccess Operation = "ListPaginatedDescendingSuccess"
+
+var listPaginatedDescendingSuccessOperation = operationTester{
+	name: OperationListPaginatedDescendingSuccess,
+	buildRequest: func(t *testing.T, testServer *httptest.Server, ct contentTypeConverter, rt ResourceTypeTest) *http.Request {
+		return buildListRequest(
+			rt.ResourceType,
+			map[string]string{
+				"take":          "2",
+				"skip":          "1",
+				"sortBy":        getSortFieldsString(rt.PaginationSortFields),
+				"sortDirection": "desc",
+			},
+			ct,
+			testServer,
+			t,
+		)
+	},
+	assertResponse: func(t *testing.T, resp *http.Response, ct contentTypeConverter, rt ResourceTypeTest) {
+		require.Equal(t, 200, resp.StatusCode)
+
+		jsonBody := responseBodyJSON(t, resp, ct)
+
+		var parsedJsonBody struct {
+			Count int              `json:count`
+			Items []map[string]any `json:items`
+		}
+		json.Unmarshal([]byte(jsonBody), &parsedJsonBody)
+
+		require.Equal(t, 3, parsedJsonBody.Count)
+
+		var prevVal any
+
+		sortFields := getSafeSortFields(rt.PaginationSortFields)
+
+		for _, field := range sortFields {
+			for _, item := range parsedJsonBody.Items {
+				if prevVal == nil {
+					prevVal = item[field]
+					continue
+				}
+				assert.True(t, compareFieldsGreaterThan(prevVal, item[field]))
```

Comment:
you can use `assert.GreaterOrEqual` so we don't have to maintain the type assertions

Reply:
It worked, thanks!